### PR TITLE
Update access revocation implementation and fix issues

### PR DESCRIPTION
**Pull request by pasin**
* Updated the replicator protocol to CBMobile_3.
* Added enable auto-purge configuration.
* Fixed issues found while testing the implementation

### CODE REVIEW

**Comment by pasin** [comment]
PR Validation failures are not related to this PR.

File: Replicator/RevFinder.cc [hunk 1]
```diff
@@ -217,6 +217,7 @@ namespace litecore::repl {
                 auto mode = (deletion < 4) ? RevocationMode::kRevokedAccess
                                            : RevocationMode::kRemovedFromChannel;
                 revoked.emplace_back(new RevToInsert(docID, mode));
+                sequences.push_back({RemoteSequence(change[0]), 0});
```
Comment (snej) [inline Replicator/RevFinder.cc:220]: This is wrong -- the `sequences` array is parallel with `docIDs` and `revIDs`. It isn't used for revoked revs.

Comment (snej) [inline Replicator/RevFinder.cc:220]: This line shouldn't be added -- the sequences array is parallel with docIDs and revIDs. It isn't used for revoked revs.

File: Replicator/RevFinder.cc [hunk 2]
```diff
@@ -250,7 +251,7 @@ namespace litecore::repl {
                 if ((status & kRevsConflict) == kRevsConflict && passive()) {
                     // Passive puller refuses conflicts:
                     encoder.writeRaw("409");
-                    sequences[i].bodySize = 0;
+                    sequences[changeIndexes[i]].bodySize = 0;
```
Comment (snej) [inline Replicator/RevFinder.cc:254]: This change isn't needed, for the reason above ^

File: Replicator/RevFinder.cc [hunk 3]
```diff
@@ -265,7 +266,7 @@ namespace litecore::repl {
             } else {
                 // I have an equal or newer revision; ignore this one:
                 // [Implicitly this appends a 0, but we're skipping trailing zeroes.]
-                sequences[i].bodySize = 0;
+                sequences[changeIndexes[i]].bodySize = 0;
```
Comment (snej) [inline Replicator/RevFinder.cc:269]: This change isn't needed, for the reason above ^

**Review by snej — CHANGES_REQUESTED** [review]

File: Replicator/RevFinder.cc (hunk 1 above)
Comment (pasin) [reply Replicator/RevFinder.cc:220]: The reason to add the sequence with body size = 0 here is to allow the checkpoint to be updated for the revoked revisions. Is there an alternative way for this?

Comment (pasin) [reply Replicator/RevFinder.cc:220]: From the Slack conversation with @snej, this solution is OK.

**Pull request merged**